File: todo-reservation-springboot/src/main/java/de/vkb/dojo/es/humanResources/model/event/PersonHealthy.java
package de.vkb.dojo.es.humanResources.model.event;

import com.fasterxml.jackson.annotation.*;

@JsonIgnoreProperties(ignoreUnknown = true)
@JsonTypeName("person-healthy")
public class PersonHealthy extends PersonEvent {
    private final String operationId;
    private final String aggregateId;

    @JsonCreator
    public PersonHealthy(
            @JsonProperty("operatonId") String operationId,
            @JsonProperty("aggregateId") String aggregateId
    ) {
        this.operationId = operationId;
        this.aggregateId = aggregateId;
    }

    @Override
    @JsonGetter
    public String getOperationId() {
        return operationId;
    }

    @Override
    @JsonGetter
    public String getAggregateId() {
        return aggregateId;
    }
}

File: mock-human-resources/src/main/kotlin/de/vkb/dojo/es/humanResources/model/event/PersonHealthy.kt
package de.vkb.dojo.es.humanResources.model.event

import com.fasterxml.jackson.annotation.JsonIgnoreProperties
import com.fasterxml.jackson.annotation.JsonTypeName

@JsonIgnoreProperties(ignoreUnknown = true)
@JsonTypeName("person-healthy")
data class PersonHealthy(
    override val operationId: String,
    override val aggregateId: String
): PersonEvent
